include/spdlog/sinks/periodic_file_sink.h

```diff
@@ -0,0 +1,217 @@
+// Copyright(c) 2015-present, Gabi Melman & spdlog contributors.
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+//File: minute_file_sink to support every 1 min file rotation.
+//example driver code:
+// auto duration{std::chrono::minutes{1}};
+//auto logger = spdlog::periodic_logger_mt("periodic_logger", "logs/min-log.txt",false,60,duration);
+
+#pragma once
+
+#include <spdlog/common.h>
+#include <spdlog/details/file_helper.h>
+#include <spdlog/details/null_mutex.h>
+#include <spdlog/fmt/fmt.h>
+#include <spdlog/sinks/base_sink.h>
+#include <spdlog/details/os.h>
+#include <spdlog/details/circular_q.h>
+#include <spdlog/details/synchronous_factory.h>
+
+#include <chrono>
+#include <cstdio>
+#include <ctime>
+#include <mutex>
+#include <string>
+
+namespace spdlog {
+namespace sinks {
+
+/*
+ * Generator of Minute log file names in format basename.YYYY-MM-DD-HH-MM.ext
+ */
+struct periodic_filename_calculator
+{
+    // Create filename for the form basename.YYYY-MM-DD-H-M
+    static filename_t calc_filename(const filename_t &filename, const tm &now_tm)
+    {
+        filename_t basename, ext;
+        std::tie(basename, ext) = details::file_helper::split_by_extension(filename);
+        return fmt_lib::format(SPDLOG_FMT_STRING(SPDLOG_FILENAME_T("{}_{:04d}-{:02d}-{:02d}-{:02d}_{:02d}{}")), basename, now_tm.tm_year + 1900, now_tm.tm_mon + 1,
+            now_tm.tm_mday, now_tm.tm_hour,now_tm.tm_min,ext);
+    }
+};
+
+/*
+ * Rotating file sink based on time(minutes).
+ * If truncate != false , the created file will be truncated.
+ * If max_files > 0, retain only the last max_files and delete previous.
+ */
+template<typename Mutex, typename FileNameCalc = periodic_filename_calculator>
+class periodic_file_sink final : public base_sink<Mutex>
+{
+public:
+   // create every periodic file sink which rotates on given time
+    periodic_file_sink(
+        filename_t base_filename, bool truncate = false, uint16_t max_files = 0,
+        std::chrono::duration<int> rotation_period = 0, const file_event_handlers &event_handlers = {})
+        : base_filename_(std::move(base_filename))
+        , file_helper_{event_handlers}
+        , truncate_(truncate)
+        , max_files_(max_files)
+        ,rotation_p_(rotation_period)
+        , filenames_q_()
+    {
+        
+          if (rotation_period <= std::chrono::minutes(0))
+        {
+            throw_spdlog_ex("periodic_file_sink: Invalid rotation time in ctor");
+        }
+        
```

Comment:
I think we should add some sanity restriction on max_files, especially since we store all the names in a buffer

```c++
  if (max_files > 2000)  
   {
         throw_spdlog_ex("periodic_file_sink: max_files param exceeds max of 2000");
   }
```

Reply:
Done